pkg/daemon/osrelease/osrelease_test.go

```diff
@@ -214,6 +214,255 @@ OSTREE_VERSION='37.20230126.20.0'`
 			assert.Equal(t, testCase.IsSCOS, os.IsSCOS(), "expected IsSCOS() to be %v", testCase.IsSCOS)
 			assert.Equal(t, testCase.IsLikeTraditionalRHEL7, os.IsLikeTraditionalRHEL7(), "expected IsLikeTraditionalRHEL7() to be %v", testCase.IsLikeTraditionalRHEL7)
 			assert.Equal(t, testCase.ToPrometheusLabel, os.ToPrometheusLabel(), "expected ToPrometheusLabel() to be %s, got %s", testCase.ToPrometheusLabel, os.ToPrometheusLabel())
+			assert.Equal(t, OSReleaseInfoSource, os.Source())
+			for key, value := range os.Values() {
+				if value != "" {
+					assert.Contains(t, testCase.OSReleaseContents, key)
+					assert.Contains(t, testCase.OSReleaseContents, value)
+				}
+			}
+		})
+	}
+}
+
+func TestOSReleaseFromImageLabels(t *testing.T) {
+	t.Parallel()
+
+	// $ skopeo inspect --no-tags "docker://$(oc adm release info --pullspecs "quay.io/openshift-release-dev/ocp-release:4.12.3-x86_64" -o json | jq -r '.references.spec.tags[] | select(.name == "rhel-coreos-8") | .from.name')" | jq '.Labels'
```

Comment:
I've included the commands I used to obtain the image labels for a sanity check. If I was looking at the wrong source image, please let me know.